flink-connector-jdbc-core/src/main/java/org/apache/flink/connector/jdbc/internal/JdbcOutputFormat.java

```diff
@@ -178,14 +178,12 @@ public synchronized void flush() throws IOException {
                     throw new IOException(e);
                 }
                 try {
-                    if (!connectionProvider.isConnectionValid()) {
-                        updateExecutor(true);
-                    }
+                    updateExecutor(!connectionProvider.isConnectionValid());
                 } catch (Exception exception) {
                     LOG.error(
-                            "JDBC connection is not valid, and reestablish connection failed.",
+                            "JDBC connection or prepared statement are not valid, and attempt to update the JDBC statement executor failed.",
```

Comment:
I think the first half can be removed.

Reply:
Do you suggest "Attempt to update the JDBC statement executor failed."?

Reply:
Yes.